R/as_gt.R

```diff
@@ -79,23 +79,37 @@ as_gt <- function(x, ...) {
 #'   summary() %>%
 #'   as_gt()
 as_gt.fixed_design <- function(x, title = NULL, footnote = NULL, ...) {
-  # get the design method
-  design_mtd <- intersect(
-    c("ahr", "fh", "mb", "lf", "rd", "maxcombo", "milestone", "rmst"), class(x)
-  )[1]
+  method <- fixed_method(x)
+  ans <- gt::gt(x) %>%
+    gt::tab_header(title = title %||% fixed_title(method)) %>%
+    gt::tab_footnote(
+      footnote = footnote %||% method_footnote(x, method),
+      locations = gt::cells_title(group = "title")
+    )
+  return(ans)
+}
+
+# get the fixed design method
+fixed_method <- function(x) {
+  methods <- c("ahr", "fh", "mb", "lf", "rd", "maxcombo", "milestone", "rmst")
+  intersect(methods, class(x))[1]
+}
 
-  # set the default title
-  if (is.null(title)) title <- sprintf("Fixed Design %s Method", switch(
-    design_mtd,
+# get the default title
+fixed_title <- function(method) {
```

Comment:
Echo with my previous comment, can we call this function as `fixed_design_title`?